File: Taller_POO/tienda_virtual.java
// Clase Cliente
class Cliente {
    int ID_cliente;
    String nombre;
    String telefono;
    String email;
    String direccion;
    String fecha_nacimiento;

    public Cliente(int ID_cliente, String nombre, String telefono, String email, String direccion, String fecha_nacimiento) {
        this.ID_cliente = ID_cliente;
        this.nombre = nombre;
        this.telefono = telefono;
        this.email = email;
        this.direccion = direccion;
        this.fecha_nacimiento = fecha_nacimiento;
    }

    public void mostrarInformacion() {
        System.out.println("Cliente: " + nombre + " (ID: " + ID_cliente + ")");
        System.out.println("Teléfono: " + telefono);
        System.out.println("Email: " + email);
        System.out.println("Dirección: " + direccion);
        System.out.println("Fecha de nacimiento: " + fecha_nacimiento);
    }

    public void comprar() {
        System.out.println(nombre + " ha realizado una compra.");
    }
}

// Clase Vendedor
class Vendedor {
    int ID_vendedor;
    String ciudad;
    String contacto;
    String tipos_productos;

    public Vendedor(int ID_vendedor, String ciudad, String contacto, String tipos_productos) {
        this.ID_vendedor = ID_vendedor;
        this.ciudad = ciudad;
        this.contacto = contacto;
        this.tipos_productos = tipos_productos;
    }

    public void mostrarInformacion() {
        System.out.println("Vendedor (ID: " + ID_vendedor + ")");
        System.out.println("Ciudad: " + ciudad);
        System.out.println("Contacto: " + contacto);
        System.out.println("Tipos de productos: " + tipos_productos);
    }

    public void confirmarPedido() {
        System.out.println("El vendedor ha confirmado el pedido.");
    }

    public void generarFactura() {
        System.out.println("El vendedor ha generado la factura.");
    }
}

// Clase Producto
class Producto {
    int ID_producto;
    String nombre_producto;
    double precio_unitario;

    public Producto(int ID_producto, String nombre_producto, double precio_unitario) {
        this.ID_producto = ID_producto;
        this.nombre_producto = nombre_producto;
        this.precio_unitario = precio_unitario;
    }

    public void mostrarInformacion() {
        System.out.println("Producto: " + nombre_producto + " (ID: " + ID_producto + ")");
        System.out.println("Precio unitario: $" + precio_unitario);
    }
}

// Clase Factura
class Factura {
    int ID_factura;
    int ID_cliente;
    int ID_vendedor;
    int ID_producto;
    double precio_total;

    public Factura(int ID_factura, int ID_cliente, int ID_vendedor, int ID_producto, double precio_total) {
        this.ID_factura = ID_factura;
        this.ID_cliente = ID_cliente;
        this.ID_vendedor = ID_vendedor;
        this.ID_producto = ID_producto;
        this.precio_total = precio_total;
    }

    public void mostrarInformacion() {
        System.out.println("Factura ID: " + ID_factura);
        System.out.println("Cliente ID: " + ID_cliente);
        System.out.println("Vendedor ID: " + ID_vendedor);
        System.out.println("Producto ID: " + ID_producto);
        System.out.println("Precio total: $" + precio_total);
    }
}

// Clase Principal
public class TiendaVirtual {
    public static void main(String[] args) {
        // Crear un cliente
        Cliente cliente1 = new Cliente(1, "María Pérez", "312345678", "maria@gmail.com", "Calle 123", "15/05/1995");
        cliente1.mostrarInformacion();
        cliente1.comprar();

        System.out.println("-------------------------");

        // Crear un vendedor
        Vendedor vendedor1 = new Vendedor(101, "Bogotá", "321987654", "Electrodomésticos");
        vendedor1.mostrarInformacion();
        vendedor1.confirmarPedido();
        vendedor1.generarFactura();

        System.out.println("-------------------------");

        // Crear un producto
        Producto producto1 = new Producto(501, "Televisor", 1500.00);
        producto1.mostrarInformacion();

        System.out.println("-------------------------");

        // Crear una factura
        Factura factura1 = new Factura(1001, cliente1.ID_cliente, vendedor1.ID_vendedor, producto1.ID_producto, producto1.precio_unitario);
        factura1.mostrarInformacion();
    }
}

File: Taller_POO/tienda_virtual.py
# -----------------------------
# Tienda Virtual
# -----------------------------

class Cliente:
    def __init__(self, ID_cliente, nombre, telefono, email, direccion, fecha_nacimiento):
        self.ID_cliente = ID_cliente
        self.nombre = nombre
        self.telefono = telefono
        self.email = email
        self.direccion = direccion
        self.fecha_nacimiento = fecha_nacimiento

    def mostrar_informacion(self):
        print(f"Cliente [{self.ID_cliente}] - {self.nombre}")
        print(f"Teléfono: {self.telefono}, Email: {self.email}")
        print(f"Dirección: {self.direccion}, Fecha de nacimiento: {self.fecha_nacimiento}")

    def comprar(self, producto):
        print(f"{self.nombre} ha comprado el producto: {producto.nombre_producto} por ${producto.precio_unitario}")


class Vendedor:
    def __init__(self, ID_vendedor, ciudad, contacto, tipos_productos):
        self.ID_vendedor = ID_vendedor
        self.ciudad = ciudad
        self.contacto = contacto
        self.tipos_productos = tipos_productos

    def mostrar_informacion(self):
        print(f"Vendedor [{self.ID_vendedor}]")
        print(f"Ciudad: {self.ciudad}, Contacto: {self.contacto}")
        print(f"Tipos de productos: {self.tipos_productos}")

    def confirmar_pedido(self, cliente, producto):
        print(f"Pedido confirmado para {cliente.nombre}: {producto.nombre_producto}")

    def generar_factura(self, ID_factura, cliente, producto):
        total = producto.precio_unitario
        return Factura(ID_factura, cliente.ID_cliente, self.ID_vendedor, producto.ID_producto, total)


class Producto:
    def __init__(self, ID_producto, nombre_producto, precio_unitario):
        self.ID_producto = ID_producto
        self.nombre_producto = nombre_producto
        self.precio_unitario = precio_unitario

    def mostrar_informacion(self):
        print(f"Producto [{self.ID_producto}] - {self.nombre_producto}, Precio: ${self.precio_unitario}")


class Factura:
    def __init__(self, ID_factura, ID_cliente, ID_vendedor, ID_producto, precio_total):
        self.ID_factura = ID_factura
        self.ID_cliente = ID_cliente
        self.ID_vendedor = ID_vendedor
        self.ID_producto = ID_producto
        self.precio_total = precio_total

    def mostrar_informacion(self):
        print(f"Factura [{self.ID_factura}]")
        print(f"Cliente ID: {self.ID_cliente}, Vendedor ID: {self.ID_vendedor}, Producto ID: {self.ID_producto}")
        print(f"Precio total: ${self.precio_total}")


# -----------------------------
# Ejemplo de uso del programa
# -----------------------------
if __name__ == "__main__":
    # Crear objetos
    cliente1 = Cliente(1, "Ana Pérez", "3124567890", "ana@mail.com", "Calle 123", "1995-04-20")
    vendedor1 = Vendedor(101, "Bogotá", "3209876543", "Ropa, Zapatos")
    producto1 = Producto(501, "Camisa Blanca", 45000)

    # Cliente compra producto
    cliente1.mostrar_informacion()
    producto1.mostrar_informacion()
    cliente1.comprar(producto1)

    # Vendedor confirma y genera factura
    vendedor1.mostrar_informacion()
    vendedor1.confirmar_pedido(cliente1, producto1)
    factura1 = vendedor1.generar_factura(9001, cliente1, producto1)

    # Mostrar factura
    factura1.mostrar_informacion()